pkg/servicenameresolutiondetector/store/node_store_test.go

```diff
@@ -0,0 +1,332 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package store
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	listcorev1 "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func TestNodeStore(t *testing.T) {
+	clientset := fake.NewSimpleClientset()
+
+	nodeInformer := cache.NewSharedIndexInformer(
+		&cache.ListWatch{},
+		&corev1.Node{},
+		0,
+		cache.Indexers{},
+	)
+	nodeLister := listcorev1.NewNodeLister(nodeInformer.GetIndexer())
+
+	store := NewNodeConditionStore(
+		clientset.CoreV1().Nodes(),
+		nodeLister,
+		string(corev1.NodeReady),
+		5*time.Minute,
+	)
+
+	testCases := []struct {
+		name           string
+		setupNodes     []*corev1.Node
+		operation      string
+		key            string
+		inputCondition *metav1.Condition
+		expectedResult interface{}
+		expectedError  bool
+	}{
+		{
+			name: "Load existing node condition",
```

Comment:
Add a case where the node exists, but the `conditionType` of the node does not match the store.